sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -1188,7 +1188,8 @@ private static void validateTimerField(
       }
 
       Class<?> stateSpecRawType = field.getType();
-      if (!(stateSpecRawType.equals(StateSpec.class))) {
+      if (!(TypeDescriptor.of(stateSpecRawType)
```

Comment:
I think it is a merge/rebase artifact. This PR was authored when `KeyedCombineFn` still existed, so it actually split `StateSpec` and `KeyedStateSpec`, so finding the superclass was necessary.

Nonetheless I think it is more correct and robust code, too, in some idealized OO sense of the word. Our design doesn't necessarily preclude `StateSpec` subclasses from being declared (except here, where it would fail).

I'll split it into a separate commit while keeping the diff the same.